rds/main.tf

```diff
@@ -0,0 +1,94 @@
+locals {
+  ports = {
+    "mysql"    = 3306
+    "postgres" = 5432
+  }
+
+  port = coalesce(var.port, local.ports[var.engine])
+}
+
+resource "random_password" "password" {
+  length           = var.random_password_length
+  override_special = "!#$%&*()-_=+[]{}<>:?" # beacuse some characters are not allowed, '@', '"', ' ', '/'
+}
+
+resource "aws_db_instance" "this" {
+  identifier     = var.identifier
+  instance_class = var.instance_class
+  engine         = var.engine
+  engine_version = var.engine_version
+
+  allocated_storage = var.allocated_storage_gb
+  storage_type      = var.storage_type
+  storage_encrypted = true
+
+  allow_major_version_upgrade = false
+  apply_immediately           = true
+  auto_minor_version_upgrade  = true
+
+  backup_retention_period = var.backup_retention_period_days
+  backup_window           = "02:00-03:00"
+  skip_final_snapshot     = var.skip_final_snapshot
+
+  deletion_protection  = var.deletion_protection
+  parameter_group_name = aws_db_parameter_group.this.name
+  publicly_accessible  = false
+
+  username = var.username
```

Comment:
lehet h nem fontos, de nem ken ennek vmi default-ot adni? rdsroot vagy ilyesmi? csak mert ez essentially az admin root user

Reply:
There is no easy way unfortunately to create a user that isn't the root. You have to create a process for that which picks up freshly created RDS instances and then creates a user that isn't the root.
For now I think it's better if one specifies the username, this will be the one the services are using.

Reply:
I know... and it's crap :(
so how do the pros do this, manually set up db users and etc? I can't imagine that running web apps with root is compliant with anything...

this is what I mentioend and had to create because there was no other usable solutions for k8s: https://github.com/bpesics/ext-postgres-op#crd-examples
maybe a similar approach could be used for rds as well...

anyway, it does not have to be solved here and now, just wanted to ask to make sure that I understand the situation...

Reply:
`eventbridge -> lambda -> create user` I guess is the AWS way.

Reply:
sure, I wast just thinking of reusing the ansible role...